src/swell/configuration/jedi/interfaces/geos_atmosphere/model/analysis.py

```diff
@@ -0,0 +1,28 @@
+# (C) Copyright 2021- United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+# --------------------------------------------------------------------------------------------------
+
+from collections.abc import Mapping
+from swell.configuration.jedi.interfaces.geos_atmosphere.model.shared import field_io_names
+
+# --------------------------------------------------------------------------------------------------
+
+
+def analysis(template_dict: Mapping) -> Mapping:
+    analysis = {
+        'filetype': 'cube sphere history',
+        'provider': 'geos',
+        'datapath': template_dict['cycle_dir'],
+        'filename': '{experiment_id}.analysis.%yyyy%mm%dd_%hh%MM%ssz.nc4'.format(**template_dict),
```

Comment:
For consistency and clarity (and to raise an `IndexError` if `experiment_id` is not set).
```suggestion
        'filename': f'{template_dict["experiment_id"]}.analysis.%yyyy%mm%dd_%hh%MM%ssz.nc4',
```